solenoid/records/models.py

```diff
@@ -201,16 +209,34 @@ def paper_requested(paper_data):
         return any([record.email.date_sent for record in records if record.email])
 
     @staticmethod
-    def is_data_valid(paper_data):
-        """Returns True if this metadata has the required data fields for
-        making a Record; False otherwise.
-
-        For citation data, we'll accept *either* a preconstructed citation,
-        *or* enough data to construct a minimal citation ourselves."""
-        citable = bool(paper_data[Fields.CITATION]) or all(
-            [bool(paper_data[x]) for x in Fields.CITATION_DATA]
-        )
-        return all([bool(paper_data[x]) for x in Fields.REQUIRED_DATA]) and citable
+    def missing_id_fields(paper_data):
+        """Check whether record is missing required ID fields."""
+        # check if record is missing ID fields ('MIT ID', 'PaperID')
+        missing_id_fields = [
+            field for field in Fields.REQUIRED_DATA if not paper_data[field]
+        ]
+        if missing_id_fields:
+            return f"Missing required ID fields: {missing_id_fields}."
+
+    @staticmethod
+    def missing_citation_fields(paper_data):
+        """Check whether record is citable.
+
+        If a citation is provided, it will be used; otherwise a minimal citation
+        is generated using the author's first name and last name and the publication's
+        title and journal (to which it was published).
+        """
+        # check if record is citable
+        missing_citation_fields = [
+            field
+            for field in [Fields.CITATION, *Fields.CITATION_DATA]
+            if not paper_data[field]
+        ]
+        if missing_citation_fields:
+            return (
+                "Citation was not provided and/or "
+                f"missing required fields to generate minimal citation: {missing_citation_fields}."
+            )
```

Comment:
Am I understanding correctly that this method will identify what minimal citation fields are missing, but that in `_run_checks_on_paper()`, it won't bail on processing the paper, it will just [log that](https://github.com/MITLibraries/solenoid/pull/472/files#diff-a02196cf0e0f7766b2a6f280c593413e90c6ddfca2295939f68560059b943ef7R109-R115)?

I ask, because the work above in `create_citation()` seems to suggest that we're somewhat okay with missing citation fields, where we just use `<Field Name UNIDENTIFIED>` and carry on.

Reply:
Yes, that is correct! Previously, when any of the checks performed in `_run_checks_on_paper()` failed, it would proceed to skip the record, which resulted in Solenoid skipping the record for import when any of the citation fields were missing.